src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +190,19 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        if action_type and action_type == "variable":
+            form_var = FormVariable.objects.get(key=variable)
+
+            if form_var.data_type == "date":
+                try:
+                    date.fromisoformat(data.get("action").get("value"))
```

Comment:
I think you can use `format_date_value` (from `openforms.utils.date`) instead of `date.fromisoformat`.

Also if `data.get("action")` returns `None`, the next get would give an error. So you would have to do `data.get("action", {}).get("value")`.